src/OpenLoco/src/World/CompanyAi.cpp

```diff
@@ -972,33 +1005,300 @@ namespace OpenLoco
+                    const auto cost = (trackBaseCost * World::TrackData::getTrackMiscData(0).costFactor) / 256;
+                    baseCost += cost;
+                }
+                {
+                    for (auto i = 0U; i < 32U; ++i)
+                    {
+                        if (thought.mods & (1U << i))
+                        {
+                            auto* trackExtraObj = ObjectManager::get<TrackExtraObject>(i);
+                            const auto trackExtraBaseCost = Economy::getInflationAdjustedCost(trackExtraObj->buildCostFactor, trackExtraObj->costIndex, 10);
+                            const auto cost = (trackExtraBaseCost * World::TrackData::getTrackMiscData(0).costFactor) / 256;
+                            baseCost += cost;
+                        }
+                    }
+                }
+                {
+                    auto* stationObj = ObjectManager::get<TrainStationObject>(thought.var_89);
+                    const auto stationBaseCost = Economy::getInflationAdjustedCost(stationObj->buildCostFactor, stationObj->costIndex, 8);
+                    const auto cost = (stationBaseCost * World::TrackData::getTrackMiscData(0).costFactor) / 256;
+                    baseCost += cost;
+
+                    if (thoughtTypeHasFlags(thought.type, ThoughtTypeFlags::unk9))
+                    {
+                        const auto tunnelBaseCost = Economy::getInflationAdjustedCost(trackObj->tunnelCostFactor, trackObj->costIndex, 10);
+                        const auto tunnelCost = (tunnelBaseCost * World::TrackData::getTrackMiscData(0).costFactor) / 256;
+                        baseCost += tunnelCost;
+                    }
+                }
+            }
+        }
+        return baseCost * costMultiplier;
+    }
+
+    // 0x00486668
+    static void sub_486668(Company& company, AiThought& thought)
+    {
+        thought.var_76 += getStationCostEstimate(thought);
+        thought.var_76 += company.var_85F2;
     }
 
     // 0x00430F50
     static void sub_430F50(Company& company, AiThought& thought)
     {
         // Calculate station cost?
-        registers regs;
-        regs.esi = X86Pointer(&company);
-        regs.edi = X86Pointer(&thought);
-        call(0x00430F50, regs);
+        if ((company.challengeFlags & CompanyFlags::unk1) != CompanyFlags::none)
+        {
+            company.var_4A4 = AiThinkState::unk6;
+            company.var_4A5 = 2;
+            company.var_85C4 = World::Pos2{ 0, 0 };
+            return;
+        }
+        sub_486668(company, thought);
+        company.var_4A5 = 5;
+    }
+
+    // 0x0048667A
+    // return
+    //   1 == No
+    //   2 == Yes
+    static uint8_t shouldConvertAiAllocated(Company& company, AiThought& thought)
+    {
+        if ((company.challengeFlags & CompanyFlags::bankrupt) != CompanyFlags::none)
+        {
+            return 1;
+        }
+
+        if (company.var_257E * 2 < thought.var_76)
```

Comment:
var_257E was the new currency you added, so this is like if the price of building the station is less than "half of the amount the AI thinks it should spend" or something?

Reply:
var_257E is the revenue estimate. So its saying if it wont be estimated to be paid off in two ?years? don't build.